extensions/sql-database-projects/src/controllers/databaseProjectTreeViewProvider.ts

```diff
@@ -0,0 +1,132 @@
+			label: element.label,
+			collapsibleState: element.parent === undefined
+				? vscode.TreeItemCollapsibleState.Expanded
+				: element.isFolder
+					? vscode.TreeItemCollapsibleState.Collapsed
+					: vscode.TreeItemCollapsibleState.None
+		};
+	}
+
+	getChildren(element?: SqlDatabaseProjectItem): SqlDatabaseProjectItem[] {
+		if (element === undefined) {
+			return this.roots;
+		}
+
+		return element.children;
+	}
+
+	public get initialized(): Promise<void> {
+		return this._initializeDeferred.promise;
+	}
+
+	async openProject(projectFiles: vscode.Uri[]) {
+		if (projectFiles.length > 1) { // TODO: how to handle opening a folder with multiple .sqlproj files?
+			vscode.window.showErrorMessage('Multiple .sqlproj files selected; please select only one.');
+		}
+
+		if (projectFiles.length === 0) {
+			vscode.window.showErrorMessage('No .sqlproj file selected; please select one.');
+		}
+
+		let directoryPath = path.dirname(projectFiles[0].fsPath);
+		console.log('Opening project directory: ' + directoryPath);
+
+		let newRoots: SqlDatabaseProjectItem[] = [];
+
+		newRoots.push(await this.constructConnectionsTree(directoryPath));
+		newRoots.push(await this.constructProjectTree(directoryPath));
+
+		this.roots = newRoots;
+		this._onDidChangeTreeData.fire();
+	}
+
+	private async constructProjectTree(directoryPath: string): Promise<SqlDatabaseProjectItem> {
+		let projectsNode = await this.constructFileTreeNode(directoryPath, undefined);
+
+		projectsNode.label = localize('projectNodeName', "Database Project");
+
+		return projectsNode;
+	}
+
+	private async constructFileTreeNode(entryPath: string, parentNode: SqlDatabaseProjectItem | undefined): Promise<SqlDatabaseProjectItem> {
+		let stat = await fs.stat(entryPath);
+
+		let output = parentNode === undefined
+			? new SqlDatabaseProjectItem(path.basename(entryPath), stat.isDirectory())
+			: parentNode.createChild(path.basename(entryPath), stat.isDirectory());
+
+		if (stat.isDirectory()) {
+			let contents = await fs.readdir(entryPath);
+
+			for (const entry of contents) {
+				await this.constructFileTreeNode(path.join(entryPath, entry), output);
+			}
+
+			// sort children so that folders come first, then alphabetical
+			output.children.sort((a: SqlDatabaseProjectItem, b: SqlDatabaseProjectItem) => {
+				if (a.isFolder && !b.isFolder) { return -1; }
+				else if (!a.isFolder && b.isFolder) { return 1; }
+				else { return a.label.localeCompare(b.label); }
+			});
+		}
+
+		return output;
+	}
+
+	private async constructConnectionsTree(directoryPath: string): Promise<SqlDatabaseProjectItem> {
+		let connectionsNode = new SqlDatabaseProjectItem(localize('connectionsNodeName', "Connections"), true);
+
+		let connectionsFilePath = path.join(directoryPath, 'connections.json');
+
+		try {
+			let connections = await fs.readFile(connectionsFilePath, 'r');
+
+			// TODO: parse connections.json
+
+			connectionsNode.createChild('Found connections.json: ' + connections.length, false);
```

Comment:
>createChild [](start = 19, length = 11)

This should be localized, and generally it's preferred to use string interpolation in cases like this (although with localize you'll just use string formatting instead)

Reply:
Yup; this was a temp string just to have a placeholder.  This will eventually be a richer set of node that allow you to configure data sources, but I'll localize it nonetheless.